frontend/catalyst/jax_primitives.py

```diff
@@ -1633,10 +1633,8 @@ def _hamiltonian_lowering(jax_ctx: mlir.LoweringRuleContext, coeffs: ir.Value, *
 def _sample_abstract_eval(obs, shots, shape):
     assert isinstance(obs, AbstractObs)
 
-    if obs.primitive is compbasis_p:
-        assert shape == (shots, obs.num_qubits)
-    else:
-        assert shape == (shots,)
+    if Signature.is_dynamic_shape(shape):
```

Comment:
The assertions cannot be kept because shots is no longer a keyword argument, it is a traceable argument. The same will happen with shape, once it is converted to traceable

Reply:
Fixed. I got rid off the `shape` argument and added a new `numqubits` arguments, building the shape inside the abstract function.